jvb/src/test/kotlin/org/jitsi/videobridge/cc/VideoConstraintsCompatibilityTest.kt

```diff
@@ -0,0 +1,28 @@
+package org.jitsi.videobridge.cc
+
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.matchers.maps.shouldContainExactly
+import org.jitsi.videobridge.VideoConstraints
+
+class VideoConstraintsCompatibilityTest : ShouldSpec() {
```

Comment:
I didn't see any difference in signaling selected/pinnned endpoints and maxFrameHeight with screensharing. The only difference seems to be that the selected endpoint is not always the dominant speaker. Am I wrong? If not, I'll add tests for the case with a selected endpooint which is not dominant speaker.

I'm actually not sure how audio-only mode is implemented. I'll check and add tests.

I am not aware of any case in which the client uses pinning. Am I wrong?

@jallamsetty1 I think you may be better able to help with some of those questions.

Reply:
I meant "pinned" in the UI sense, i.e. when you explicitly click on someone in stage view rather than letting it be by dominant speaker, rather than meaning the signaling message.

Probably this isn't different for the CompatibilityTest, but it is for the BitrateController proper.

Reply:
For client-only mode, we set lastN to 0. When a share is detected while the client is in audio-only mode, lastN is set to 1 and the share is automatically pinned.
I think desktop share and youtube share are the only sources which are automatically pinned by the client.

Reply:
Great, thanks for the input! I added:
1. Tests for the "pinned in the UI" case (i.e. non-dominant speaker is on stage)
2. Tests for audio-only mode (i.e. LastN=0)
3. Tests for audio-only + screenshare (i.e. LastN=1 + a selected endpoint, possibly not the dominant speaker).

I also discovered I was not setting the `eid` of layers correctly, which explains at least some of the weird behavior.